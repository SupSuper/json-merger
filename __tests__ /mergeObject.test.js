const jsonMerger = require("../dist");
const {testConfig} = require("./__helpers__");

describe(".mergeObject() should process the object and return", function () {

    test("the processed object if config.operationPrefix is @", function () {

        const object = {
            "a": {
                "@replace": 1,
            },
            "$replace": 2
        };

        const result = jsonMerger.mergeObject(object, testConfig({
            operationPrefix: "@"
        }));

        expect(result).toMatchSnapshot();
    });

    test("the processed object if config.stringify is false", function () {

        const object = {
            "a": {
                "$replace": 1
            }
        };

        const result = jsonMerger.mergeObject(object, testConfig({
            stringify: false
        }));

        expect(result).toMatchSnapshot();
    });

    test("a JSON string if config.stringify is true", function () {

        const object = {
            "a": {
                "$replace": 1
            }
        };

        const result = jsonMerger.mergeObject(object, testConfig({
            stringify: true
        }));

        expect(result).toMatchSnapshot();
    });

    test("a pretty JSON string if config.stringify is 'pretty'", function () {

        const object = {
            "a": {
                "$replace": 1
            }
        };

        const result = jsonMerger.mergeObject(object, testConfig({
            stringify: "pretty"
        }));

        expect(result).toMatchSnapshot();
    });
});
