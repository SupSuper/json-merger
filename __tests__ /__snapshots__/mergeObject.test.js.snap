// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.mergeObject() should process the object and return a JSON string if config.stringify is true 1`] = `"{\\"a\\":1}"`;

exports[`.mergeObject() should process the object and return a pretty JSON string if config.stringify is 'pretty' 1`] = `
"{
	\\"a\\": 1
}"
`;

exports[`.mergeObject() should process the object and return the processed object if config.operationPrefix is @ 1`] = `
"{
	\\"a\\": 1,
	\\"$replace\\": 2
}"
`;

exports[`.mergeObject() should process the object and return the processed object if config.stringify is false 1`] = `
Object {
  "a": 1,
}
`;
